src/TLuaInterpreter.cpp

```diff
@@ -12083,6 +12151,75 @@ void TLuaInterpreter::initLuaGlobals()
     //FIXME make function call in destructor lua_close(L);
 }
 
+// initialised a slimmed-down Lua state just to run the indenter in a separate sandbox
+// the indenter by default pollutes the global environment with some utility functions
+// and we don't want to tie ourselves to it by exposing them for scripting
+void TLuaInterpreter::initIndenterGlobals()
+{
+    pIndenterState = newstate();
+    storeHostInLua(pIndenterState, mpHost);
+
+    luaL_openlibs(pIndenterState);
+
+    lua_pushstring(pIndenterState, "SESSION");
+    lua_pushnumber(pIndenterState, mHostID);
+    lua_settable(pIndenterState, LUA_GLOBALSINDEX);
+
+    lua_pushstring(pIndenterState, "SCRIPT_NAME");
+    lua_pushstring(pIndenterState, "Lua Indenter Interpreter");
+    lua_settable(pIndenterState, LUA_GLOBALSINDEX);
+
+    lua_pushstring(pIndenterState, "SCRIPT_ID");
+    lua_pushnumber(pIndenterState, -2); // ID 2 is used to indicate that this is the indenter Lua interpreter
+    lua_settable(pIndenterState, LUA_GLOBALSINDEX);
+    lua_register(pIndenterState, "echo", TLuaInterpreter::Echo);
+    lua_register(pIndenterState, "tempTimer", TLuaInterpreter::tempTimer);
+    lua_register(pIndenterState, "send", TLuaInterpreter::sendRaw);
+    lua_register(pIndenterState, "debugc", TLuaInterpreter::debug);
+    // PLACEMARKER: End of Lua functions registration
+
+    luaopen_yajl(pIndenterState);
+    lua_setglobal(pIndenterState, "yajl");
+
+    QString n;
+    int error;
+
+#ifdef Q_OS_LINUX
+    // if using LuaJIT, adjust the cpath to look in /usr/lib as well - it doesn't by default
+    luaL_dostring(pIndenterState, "if jit then package.cpath = package.cpath .. ';/usr/lib/lua/5.1/?.so;/usr/lib/x86_64-linux-gnu/lua/5.1/?.so' end");
+
+    //AppInstaller on Linux would like the search path to also be set to the current binary directory
+    luaL_dostring(pIndenterState, QString("package.cpath = package.cpath .. ';%1/lib/?.so'").arg(QCoreApplication::applicationDirPath()).toUtf8().constData());
+#endif
+#ifdef Q_OS_MAC
+    //macOS app bundle would like the search path to also be set to the current binary directory
+    luaL_dostring(pGlobalLua, QString("package.cpath = package.cpath .. ';%1/?.so'").arg(QCoreApplication::applicationDirPath()).toUtf8().constData());
+#endif
+
+
+    error = luaL_dostring(pIndenterState, R"(
+      require('lcf.workshop.base')
+      get_ast = request('!.lua.code.get_ast')
+      get_formatted_code = request('!.formats.lua.save')
+    )");
+    if (error != 0) {
+        string e = "no error message available from Lua";
+        if (lua_isstring(pIndenterState, -1)) {
+            e = "Lua error:";
```

Comment:
I think the "Lua error:" needs a space to be appended so that there is a unit of white-space between the `:` and the first word in the error message - which is a code pattern error in the other places where this was copied from! 🙁